sql/auth/sql_user_table.cc

```diff
@@ -2040,6 +2014,30 @@ int open_grant_tables(THD *thd, TABLE_LIST *tables,
   return 0;
 }
 
+#ifdef WITH_WSREP
+bool start_toi_after_open_grant_tables(THD *thd, const char *db,
```

Comment:
Can't this result in a deadlock with unlucky timing? (both nodes lock at the same time, and try to start TOI after that)

Reply:
The sequence is:
1. Lock tables
2. Start TOI

If node_1 is between (1) and (2), and node_2 executes TOI, the operation on node_1 will fail (BF-abort).
I've added the test case for this.